contrib/babelfishpg_tsql/src/procedures.c

```diff
@@ -2201,50 +2209,107 @@ gen_sp_droprolemember_subcmds(const char *user, const char *member)
+	{
+		int32		query_timeout;
+		
+		if (optvalue == NULL || strspn(optvalue, "0123456789") != strlen(optvalue))
+			ereport(ERROR,
+					(errcode(ERRCODE_FDW_ERROR),
+					 errmsg("Invalid option value for query timeout")));
+		else
+			query_timeout = atoi(optvalue);
+
+		if (query_timeout < 0)
+			ereport(ERROR,
+					(errcode(ERRCODE_FDW_ERROR),
+					 errmsg("Query timeout value provided is out of range")));
 
-	parsetree_list = raw_parser(query_str, RAW_PARSE_DEFAULT);
+		bbf_servers_def_rel = table_open(get_bbf_servers_def_oid(),
+									 	RowExclusiveLock);
+		bbf_servers_def_rel_dsc = RelationGetDescr(bbf_servers_def_rel);
 
-	if (list_length(parsetree_list) != 1)
-		ereport(ERROR,
-				(errcode(ERRCODE_SYNTAX_ERROR),
-				 errmsg("Expected 1 statement but get %d statements after parsing",
-						list_length(parsetree_list))));
-
-	/* Update the dummy statement with real values */
-	stmt = parsetree_nth_stmt(parsetree_list, 0);
-
-	/* Run the built query */
-	/* need to make a wrapper PlannedStmt */
-	wrapper = makeNode(PlannedStmt);
-	wrapper->commandType = CMD_UTILITY;
-	wrapper->canSetTag = false;
-	wrapper->utilityStmt = stmt;
-	wrapper->stmt_location = 0;
-	wrapper->stmt_len = strlen(query_str);
-
-	/* do this step */
-	ProcessUtility(wrapper,
-				   query_str,
-				   false,
-				   PROCESS_UTILITY_SUBCOMMAND,
-				   NULL,
-				   NULL,
-				   None_Receiver,
-				   NULL);
-
-	/* make sure later steps can see the object created here */
-	CommandCounterIncrement();
+		MemSet(new_record_nulls, false, sizeof(new_record_nulls));
+		MemSet(new_record_repl, false, sizeof(new_record_repl));
+
+		new_record[Anum_bbf_servers_def_servername - 1] = CStringGetTextDatum(servername);
+		new_record[Anum_bbf_servers_def_query_timeout - 1] = Int32GetDatum(query_timeout);
+
+		if(isInsert)
+		{
+			tuple = heap_form_tuple(bbf_servers_def_rel_dsc,
+									new_record, new_record_nulls);
+			CatalogTupleInsert(bbf_servers_def_rel, tuple);
+		}
+		else
+		{
+			/* Search and obtain the tuple based on the server_id */	
+			ScanKeyInit(&key,
+						Anum_bbf_servers_def_servername,
+						BTEqualStrategyNumber, F_TEXTEQ,
+						CStringGetTextDatum(servername));
+			tblscan = table_beginscan_catalog(bbf_servers_def_rel, 1, &key);
+			old_tuple = heap_getnext(tblscan, ForwardScanDirection);
+
+			if (!old_tuple)
+			{
+				table_endscan(tblscan);
+				table_close(bbf_servers_def_rel, RowExclusiveLock);
+				ereport(ERROR,
+						(errcode(ERRCODE_FDW_ERROR),
+				 		errmsg("server \"%s\" does not exist", servername)));
+			}
+			new_record_repl[Anum_bbf_servers_def_query_timeout - 1] = true;
+			tuple = heap_modify_tuple(old_tuple, bbf_servers_def_rel_dsc,
+									new_record, new_record_nulls, new_record_repl);
+			
+			CatalogTupleUpdate(bbf_servers_def_rel, &tuple->t_self, tuple);
+			table_endscan(tblscan);
+		}
+
+		heap_freetuple(tuple);
+
+		table_close(bbf_servers_def_rel, RowExclusiveLock);
+	}
```

Comment:
add else block and throw same error "Invalid option provided for sp_serveroption"